libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java

```diff
@@ -277,13 +278,14 @@ public void showAlternativeRoutes(boolean alternativesVisible) {
    * This method will allow this class to listen to route progress and adapt the route line
    * whenever {@link TripSessionState#STARTED}.
    *
+   * In order to use the vanishing route line feature be sure to enable the feature before
+   * calling this method.
```

Comment:
I think it's just the wording but wanted to confirm - if we call `addProgressChangeListener` and then `enableVanishingRouteLine`, the feature will still be enabled, right? If yes, how about:
```suggestion
   * In order to use the vanishing route line feature be sure to call {@link #enableVanishingRouteLine()}.
```

Reply:
No. The progress change listener gets created differently depending on whether the the vanishing route line is enabled.  For example no animator will get created if the feature is disabled. Previously this condition was covered by the builder which took MapboxNavigation and a boolean for the vanishing route line as arguments.  That was/will be removed for the newest iteration of the handling of the vanishing route line. So to update the API I changed the builder.  It's not obvious though that the progress change listener is dependent on the vanishing route line being enabled first.

Reply:
Couldn't we recreate progress listener after `enableVanishingRouteLine` is called?